app/models/account/provider_methods.rb

```diff
@@ -68,7 +68,7 @@ def build_kind(kind:, **attributes)
       end
     end
 
-    has_many :buyer_accounts, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
+    has_many :buyer_accounts, -> { where(Account.arel_table[:id].not_eq(Account.arel_table[:provider_account_id])) }, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
```

Comment:
```suggestion
    has_many :buyer_accounts, ->(provider) { where.not(id: provider.id) }, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
```

Reply:
How about this?

I think it's more concise and more readable/understandable. And produces the same result.

I actually thought that it might be more efficient, because it search in the DB using indexes and primary keys directly, while the `id <> provider_account_id` would actually compare the values for each record, but the DB is smarter and results in the very similar query (from what I can tell).

Reply:
In the Arel version the query is:
```
[34] pry(main)> account.buyer_accounts.count
   (2.3ms)  SELECT COUNT(*) FROM `accounts` WHERE `accounts`.`provider_account_id` = 2 AND `accounts`.`id` != `accounts`.`provider_account_id`
```

In your suggested version the query is:
```
[2] pry(main)> account.buyer_accounts.count
   (1.8ms)  SELECT COUNT(*) FROM `accounts` WHERE `accounts`.`provider_account_id` = 2 AND `accounts`.`id` != 2
```

Searching by primary index is not possible nor advisable because the huge work is getting the buyer ids by the index on `provider_account_id`. Then the additional filtering cannot use an index. Either case, I assume the engine is smart enough to treat the same negation and "not equal" the same.

But comparing a constant should apparently be slightly better than comparing to another field.

It is very different semantics though - filtering out self vs filtering out id/provider_id matches. I like your suggstion much better but it initially confused me of why it worked. So I accept change but rename `provider` to `this_account`. 

Does it look good to you that way?

Reply:
The logic behind `this_account` is that rails seems to pass current object to the scope lambda. So we filter out buyers that have an `id` same as current object id. I'm not sure anymore what makes more sense. I'd rather update it to `_1` numbered parameter once on Ruby 2.7. WDYT?

Reply:
Well, yeah, sorry, not a full table scan definitely.... :facepalm:  
Just got confused for a bit.

Reply:
@akostadinov What do you mean by `_1` for Ruby 2.7? I don't understand that part

Reply:
Better don't use that `_1` pattern IMO. It's harder to understand than `provider`.

Reply:
I think it is very useful when there is one thing that is passed by convention or by usage and especially when a good short term is unavailable to describe it. But it's easy not to touch after merge, so your wish will be granted :)